src/main/java/de/tum/in/www1/artemis/domain/participation/TeamParticipation.java

```diff
@@ -0,0 +1,133 @@
+package de.tum.in.www1.artemis.domain.participation;
+
+import java.util.Optional;
+
+import javax.persistence.*;
+
+import org.hibernate.Hibernate;
+
+import com.fasterxml.jackson.annotation.JsonView;
+
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.modeling.ModelingSubmission;
+import de.tum.in.www1.artemis.domain.team.Team;
+import de.tum.in.www1.artemis.domain.view.QuizView;
+
+@Entity
+@DiscriminatorValue(value = "TP")
+public class TeamParticipation extends Participation {
+
+    private static final long serialVersionUID = 1L;
+
+    @Column(name = "presentation_score")
+    private Integer presentationScore;
+
+    @ManyToOne
+    @JsonView(QuizView.Before.class)
+    private Team team;
+
+    public Integer getPresentationScore() {
+        return presentationScore;
+    }
+
+    public TeamParticipation presentationScore(Integer presentationScore) {
+        this.presentationScore = presentationScore;
+        return this;
+    }
+
+    public void setPresentationScore(Integer presentationScore) {
+        this.presentationScore = presentationScore;
+    }
+
+    public Team getTeam() {
+        return team;
+    }
+
+    public Participation team(Team team) {
+        this.team = team;
+        return this;
+    }
+
+    public void setTeam(Team team) {
+        this.team = team;
+    }
+
+    public Exercise getExercise() {
+        return exercise;
+    }
+
+    public TeamParticipation exercise(Exercise exercise) {
+        this.exercise = exercise;
+        return this;
+    }
+
+    public void setExercise(Exercise exercise) {
+        this.exercise = exercise;
+    }
+
+    /**
+     * Removes the team from the participation, can be invoked to make sure that sensitive information is not sent to the client. E.g. tutors should not see information about
+     * the team.
+     */
+    public void filterSensitiveInformation() {
+        setTeam(null);
+    }
+
+    private <T extends Submission> Optional<T> findLatestSubmissionOfType(Class<T> submissionType) {
```

Comment:
I was able to avoid the duplication by introducing a new abstract class `ParticipantParticipation`.

The name seems a bit forced but I couldn't think of anything better that would capture the essence of both `StudentParticipation` and `TeamParticipation` but does not include e.g. `TutorParticipation`. Alternatives that I dismissed were `UserParticipation` (would include `TutorParticipation`, `StudentsParticipation` (too similar to `StudentParticipation`, could cause confusion), `BaseStudentParticipation`, `AbstractStudentParticipation` (the last two would be feasible in my opinion but this kind of naming scheme isn't really used in the codebase so far). Let me know if you have a better idea regarding the naming.

`ParticipantParticipation` includes all the shared code while the subclasses implement the differences.